### ISSUE TITLE
save column visibilities server side not client side

### ISSUE THREAD

**Issue by j0hannesr0th** [kevinpapst/kimai2#1230]
**Is your feature request related to a problem? Please describe.**
At the moment column visibilities are saved client side:
![grafik](https://user-images.githubusercontent.com/36242595/68426553-a7d35580-01a8-11ea-9213-ef05c88227f5.png)

It would be great if this could be saved server side!

**Comment by kevinpapst** [comment]
Pro: consistent
Con: terrible experience if you switch between desktop and mobile

**Comment by hmr-it-jr** [comment]
What about saving it per user-agent or for desktop and mobile.

I don't think mobile views would change because at the moment if you want the change the visibility on mobile devices it doesn't work. So all columns are selected but only some are visible on mobile devices.

**Comment by kevinpapst** [comment]
Then you have to save the changes again after a browser update/change?
Not that I donÄt want to work on it, it just needs to be bullet proof.

For me it works on mobile, you just have to reload the page after you change the visibility.
There is "just" a minor bug, that the current hidden columns are pre-selected on mobile. 
I thought I had fixed that, but apparently there is still room for improvement ;-)

**Comment by hmr-it-jr** [comment]
If I click on the visibility, select everything (which is default) and hit save, reload page all columns could be visible, but I'm not able to swipe left or right to see the new visible columns.
Tested on Android 9 in Chrome and FF.

No, the default visibility should be stored in database - and if you want to separate mobile and desktop im fine with that.

**Comment by kevinpapst** [comment]
True, there is a custom css rule to activate swiping on the tables.

> and if you want to separate mobile and desktop im fine with that.

If you have good idea how, please let me know!

**Comment by hmr-it-jr** [comment]
Maybe with this lib: http://mobiledetect.net/

```
// Any mobile device (phones or tablets).
if ( $detect->isMobile() ) {
 
}
```

**Comment by kevinpapst** [comment]
I am closing this, because it is available in the version which will be released as 2.0 in the future.

If you want to help testing and provide feedback, please follow the issue https://github.com/kevinpapst/kimai2/pull/2902.

The changes can be tested at https://demo-branch.kimai.org with the demo logins (e.g. `susan_super` and `kitten`).
